pkgs/development/php-packages/relay/default.nix

```diff
@@ -0,0 +1,113 @@
+{ stdenv
+, lib
+, fetchurl
+, php
+, openssl
+, openssl_1_1
+, zstd
+, lz4
+, autoPatchelfHook
+}:
+
+let
+  version = "0.6.3";
+  system = stdenv.hostPlatform.system;
+  phpVersion = lib.versions.majorMinor php.version;
+  variation = {
+    "aarch64-darwin" = {
+      platform = "darwin-arm64";
+      hashes = {
+        "8.0" = "00a7pyf9na7hjifkmp2482c7sh086w72zniqgr4cz2rhz7hnqp7p";
+        "8.1" = "0mg6nsllycgjxxinn8s30y9sk06g40vk8blnpx0askjw5zdsf5y7";
+        "8.2" = "0qmcbrj6jaxczv25rdgfjrj9nwq4vb2faw7kzlyxrvvzb5pyn9dm";
+        "8.3" = "1hqjy5y4q3alxvrj7xksaf7vvmz8p51bgzxbvmzdx6jnl63dix33";
+      };
+    };
+    "aarch64-linux" = {
+      platform = "debian-aarch64+libssl3";
+      hashes = {
+        "8.0" = "19zzw4324z096b6bph1686r30i4i2kwmlmmcqmb33lqkr9b9n5ag";
+        "8.1" = "0j6wpwy8d67pqij4v8m2xwydfddzr7nn4c3lyrssp8llbn4ghwpn";
+        "8.2" = "1nbajvi5zk6z8qr32l86p65f1zxv12kald56pg8k7bj4axlj2pmy";
+        "8.3" = "1sn638g2636m6s3lv2cclza9lzmzgqxamcga7jz3ijhn2ja6znbv";
+      };
+    };
+    "x86_64-darwin" = {
+      platform = "darwin-x86-64";
+      hashes = {
+        "8.0" = "13q6va9lxgfyx86xw20iqjz4s9r9xms74ywb2hgqrhs5mjnazzz4";
+        "8.1" = "1ncih5bf0jcylpl0nj8hi50kcwb4nl1g0ql359dgg9gp8s1b4hmw";
+        "8.2" = "150difm3vg0g2pl5hb5cci4jzybd7jcd7prjdv3rmwsi91gsydlv";
+        # 8.3 for this platform does not exist
+      };
+    };
+    "x86_64-linux" = {
+      platform = "debian-x86-64+libssl3";
+      hashes = {
+        "8.0" = "1qvd5r591kp7qf9pkymz78s8llzs1vxjwqhwy9rvma21hh6gd8ld";
+        "8.1" = "0fm9ppgx7qaggid134qnl9jkq5h97dac2ax21f1f1d0k8f5blmc2";
+        "8.2" = "0g2yqwfrigf047dqkrvfcj318lvgp38zy522ry484mrgq7iqwvb8";
+        "8.3" = "06mi3yr7k1a9icdljzl76xvrw6xqnvwiavgzx2g487s097mycjp4";
+      };
+    };
+  }.${system} or (throw "Unsupported platform for relay: ${system}");
+in
+stdenv.mkDerivation {
```

Comment:
I don't know if it would be too much to implement the `finalAttrs` pattern? See https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes

Reply:
I think so, I don't see a fit here. This is in order to customize which attribute? There are 17 PHP packages utilizing mkDerivation, but only grumphp is using this.

Reply:
`finalAttrs` is a good practice, but it's totally not applied all the time.

@jtojnar Is `finalAttrs`  something that we must do or it can be optional ?

Reply:
The benefit would be that you could override `version` in the `url` without  having to replace whole URL.

There are no rules requiring it but it is a good to have in all new packages.